### ISSUE TITLE
Custom Object Detection Model - My model doesn't work on my own dataset

### ISSUE THREAD

**Issue by marbortoli** [OlafenwaMoses/ImageAI#711]
My model comes out with good accuracy after 100 epochs, however, when to detect the objects in a random image, nothing is detected. 
Someone here really used ImageAI on your own dataset?
Is there some limitation according to image resolution?

Cheers

**Comment by ekesdf** [comment]
Hi, 

Can you give me some details about your custom dataset. So I can get a better view about the situation.

**Comment by marbortoli** [comment]
_import imageai
from imageai.Detection.Custom import DetectionModelTrainer
# train the custom model
trainer = DetectionModelTrainer()
trainer.setModelTypeAsYOLOv3()
trainer.setDataDirectory(data_directory='scada_all')
trainer.setTrainConfig(object_names_array=["tag"],
                       batch_size=4,
                       num_experiments= 100,
                       train_from_pretrained_model=(\bla\pretrained-yolov3.h5)
train = trainer.trainModel()
print(train)
# evaluation of the model

metrics = trainer.evaluateModel(model_path='scada_all/models', 
                               json_path='scada_all/json/detection_config.json', 
                               iou_threshold=0.5, 
                               object_threshold=0.3,
                               nms_threshold=0.5)
print(metrics)_

my model loss = 13.616 and mAP =~ 80%

_# image detection
detector = CustomObjectDetection()
detector.setModelTypeAsYOLOv3()
detector.setModelPath('scada_all/models/detection_model-exBLABLABLA') 
detector.setJsonPath('scada_all/json/detection_config.json')
detector.loadModel()
detections = detector.detectObjectsFromImage(input_image=input_path, output_image_path=output_path)
for detection in detections:
    print(detection['name'], ' : ', detection['percentage_probability'], ' : ', detection['box_points'])_
  

![single_sensor_original_24 png_2f77a9a4-fc77-4495-be57-ee7c388a4cab](https://user-images.githubusercontent.com/86786056/139698806-69079c54-11c4-46d1-84dd-16def2e3df5f.png)
 
my dataset is made of images like this, where I want to detect one of this tags. I did data augmentation to increase the amount of images to train the model.

**Comment by ekesdf** [comment]
how big are your images  compared to the objects you want to detect ?

**Comment by marbortoli** [comment]
I have different styles of images. I'm not sure if the quality is too low.
![single_sensor_original_01 png_8b1987ac-0328-4083-832d-2ee6c6bae173](https://user-images.githubusercontent.com/86786056/139700424-0b30ee6a-e4dc-46e0-a929-1ecbc180218b.png)

![single_sensor_original_07 png_f99f4483-84cd-4af0-8395-fab91caa3cf1](https://user-images.githubusercontent.com/86786056/139700441-64843c0f-1014-47f3-b080-c7bfdcff5579.png)

![single_sensor_original_08 png_99218f01-25a3-4b06-b25b-58d802c19620](https://user-images.githubusercontent.com/86786056/139700575-bba134e3-83c4-484f-9e3e-bb9c7c10158a.png)

And I would detect in a picture very similar, like this:
![Capture](https://user-images.githubusercontent.com/86786056/139700541-b25c1a53-3047-446e-b848-d7d5540c244c.PNG)

**Comment by ekesdf** [comment]
How big  is the resolution in like pixels

**Comment by marbortoli** [comment]
it varies like in this examples, from 76x57 to 1037x612

**Comment by ekesdf** [comment]
This range should not be a problem. The problem might be you labels itself bc they have a lot of background so the

**Comment by ekesdf** [comment]
It should be more like this.
![139700541-b25c1a53-3047-446e-b848-d7d5540c244c (1) (Kopie)](https://user-images.githubusercontent.com/64316659/139705092-2fc874dc-48e7-4887-a114-6cc26b66bcd9.png)

**Comment by ekesdf** [comment]
If those 3 images you send me where cut our labels then are the objects in them are too small compared to the background.
They should look more like this.
![139700541-b25c1a53-3047-446e-b848-d7d5540c244c (1) (Kopie)](https://user-images.githubusercontent.com/64316659/139705388-26662ac8-d67d-49c1-b4c1-b7f8ede9cb94.png)

**Comment by marbortoli** [comment]
Thanks a lot! The object detection  model works, but not for every input image I provide. I'll try to improve my dataset. 

Thanks a lot again!!

**Comment by ekesdf** [comment]
@marbortoli 
Hey how it is going could you fix your problem ?

**Comment by marbortoli** [comment]
I could fix in general. The main issue now is that the model doesn't work for every image I give as an input. I'm increasing the epochs to try to solve the problem :)

**Comment by ekesdf** [comment]
That is the problem regarding to the object detecting using machine learning that the accuracy will never be 100 %
Btw if u train the model to long it could decrease the acc on images outside you training se. So if the loss is stable on a low lvl for a long time you should stop training to prevent that

**Comment by marbortoli** [comment]
Thanks! 
So far my accuracy is very low anyways. I don't know if its due to my dataset, but I provided a lot of images. However many of them are from data augmentation and I don't know if this can compromise my training in some way.

**Comment by ekesdf** [comment]
What kind of augmentation do u use ?

**Comment by marbortoli** [comment]
Augmentor library from python

**Comment by ekesdf** [comment]
@marbortoli 
hi can u specify exactly what u used things like rotation or random cropping ?

**Comment by marbortoli** [comment]
oh sorry for the delay.

p.rotate(probability=0.9, max_left_rotation=10, max_right_rotation=10)
p.zoom(probability=0.3, min_factor=1.1, max_factor=1.6)
